src/main/java/com/gmail/goosius/siegewar/SiegeController.java

```diff
@@ -628,10 +632,14 @@ public static void beginSiegeCamp(SiegeCamp camp) throws TownyException {
 		if (SiegeCampUtil.hasFailedCamp(camp.getTargetTown(), camp.getTownOfSiegeStarter()))
 			throw new TownyException(Translatable.of("msg_err_too_soon_since_your_last_siegecamp"));
 
+		Translatable translatable = Translatable.of("attacker_has_begun_a_siegecamp_session", camp.getTownOfSiegeStarter(), camp.getTargetTown(), SiegeWarSettings.getSiegeCampPointsForSuccess(), SiegeWarSettings.getSiegeCampDurationInMinutes());
+
 		// Broadcast a message
-		Messaging.sendGlobalMessage(Translatable.of("attacker_has_begun_a_siegecamp_session", camp.getTownOfSiegeStarter(), camp.getTargetTown(), SiegeWarSettings.getSiegeCampPointsForSuccess(), SiegeWarSettings.getSiegeCampDurationInMinutes()));
+		Messaging.sendGlobalMessage(translatable);
 		// Add to SiegeCamp list and begin Evaluating this SiegeCamp for success.
 		addSiegeCamp(camp);
 		SiegeWar.getSiegeWar().getScheduler().run(camp.getBannerBlock().getLocation(), ()-> SiegeCampUtil.evaluateCamp(camp, true));
+		// Call event
+		Bukkit.getPluginManager().callEvent(new SiegeCampStartEvent(camp, translatable.toString()));

```

Comment:
instead of doing translatable.toString() you should probably use `translatable.defaultLocale();`